### PR TITLE
[WIP] initial fix of socket streaming by guessing playback time

### LINKED ISSUE #2945: Audio stream internal data stream error when using socket streaming
### Bug Report
Audio stream returns a "internal data stream" error when using gstreamer command for socket streaming

##### Reproduction Steps
1. Set `AudioStreamConsumer` to `socket` and `AudioStreamingPort` to `5080` in `smartDeviceLink.ini`.
2. Start audio stream from app (accept prompt from HMI to start audio stream)
3. Start reading audio stream using the following `gstreamer` command:
```
gst-launch-1.0 souphttpsrc location=http://127.0.0.1:5080 ! audio/x-raw,format=S16LE,rate=16000,channels=1 ! pulsesink
```

##### Expected Behavior
Entire audio file is played

##### Observed Behavior
No audio is played and the audio stream is stopped due to an internal data stream error

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: 5.1.2
* Testing Against: sdl_test_suite

##### Test Case
The `gstream` command prints the following output 
```
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
Got context from element 'souphttpsrc0': gst.soup.session=context, session=(SoupSession)NULL, force=(boolean)false;
ERROR: from element /GstPipeline:pipeline0/GstSoupHTTPSrc:souphttpsrc0: Internal data stream error.
Additional debug info:
gstbasesrc.c(3055): gst_base_src_loop (): /GstPipeline:pipeline0/GstSoupHTTPSrc:souphttpsrc0:
streaming stopped, reason error (-5)
ERROR: pipeline doesn't want to preroll.
Setting pipeline to NULL ...
Freeing pipeline ...
```

### PR DESCRIPTION
**Pull request by iCollin**
Partially Fixes #2945 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF, Manually

### Summary
Add logic for Core to guess how long an audio stream will take based on PCM capabilities.

##### Enhancements
* Socket streaming somewhat works

### Tasks Remaining:
- [x] Decide how much latency compensation should be added in DataSizeToMilliseconds
- [x] Restrict logic in this PR to only applicable scenarios (streaming is socket, maybe just for audio)
- [x] Test a lot more!
- [x] Make sure it builds with tests, add mock functions!

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/include/application_manager/application.h [hunk 1]
```diff
@@ -657,7 +657,8 @@ class Application : public virtual InitialApplicationData,
    * @brief Wakes up streaming process for application
    * @param service_type Type of streaming service
```
Comment (ShobhitAd) [inline src/components/application_manager/include/application_manager/application.h]: Probably add a `@param` line in the function brief explaining what `timer_len` is

File: src/components/include/media_manager/media_manager.h [hunk 2]
```diff
@@ -60,6 +60,8 @@ class MediaManager {
    */
   virtual const MediaManagerSettings& settings() const = 0;
 
+  virtual uint32_t DataSizeToMilliseconds(uint64_t data_size) const = 0;
```
Comment (ShobhitAd) [inline src/components/include/media_manager/media_manager.h]: Might be good to add a function brief  for this function

File: src/components/media_manager/src/media_manager_impl.cc [hunk 3]
```diff
@@ -160,6 +164,23 @@ void MediaManagerImpl::Init() {
     streamer_[ServiceType::kAudio]->AddListener(
         streamer_listener_[ServiceType::kAudio]);
   }
+
+  if (application_manager_.hmi_capabilities().pcm_stream_capabilities()) {
+    auto pcm_caps =
```
Comment (ShobhitAd) [inline src/components/media_manager/src/media_manager_impl.cc]:
```suggestion
    const auto& pcm_caps =
```

File: src/components/include/media_manager/media_manager.h (hunk 2 above)
Comment (iCollin) [reply src/components/include/media_manager/media_manager.h]: Added in https://github.com/smartdevicelink/sdl_core/pull/3057/commits/c61c05654ee2d4919b03e0e539eb86ba030f3aba

File: src/components/application_manager/include/application_manager/application.h (hunk 1 above)
Comment (iCollin) [reply src/components/application_manager/include/application_manager/application.h]: Added in https://github.com/smartdevicelink/sdl_core/pull/3057/commits/c61c05654ee2d4919b03e0e539eb86ba030f3aba

File: src/components/media_manager/src/media_manager_impl.cc (hunk 3 above)
Comment (iCollin) [reply src/components/media_manager/src/media_manager_impl.cc]: Added (but not as ref cause its a pointer) in https://github.com/smartdevicelink/sdl_core/pull/3057/commits/c61c05654ee2d4919b03e0e539eb86ba030f3aba

**Pull request merged**